File: freight-triage-polyglot-ledger-recovery/environment/intake/src/com/freight/stats/StddevMilli.java
package com.freight.stats;

/** stddev_milli kernel. */
public final class StddevMilli implements StatKernel {

    @Override
    public String name() {
        return "stddev_milli";
    }

    @Override
    public long apply(long[] series) {
        long variance = new VarianceMilli().apply(series);
        return StatSupport.integerSqrt(variance * 1000L);
    }
}

File: freight-triage-polyglot-ledger-recovery/environment/native/src/stats/stddev_milli.cpp
#include "freight/stats.h"

#include <algorithm>

namespace freight {

// stddev_milli kernel.
long long stat_stddev_milli(const std::vector<long long>& series) {
  long long variance = stat_variance_milli(series);
  return statIntegerSqrt(variance * 1000);
}

}  // namespace freight
